tests/test_data_handling.py

```diff
@@ -0,0 +1,105 @@
+"""
+SPDX-FileCopyrightText: 2019 oemof developer group <contact@oemof.org>
+SPDX-License-Identifier: MIT
+"""
+
+import os
+from datetime import datetime
+from shutil import copyfile
+
+import pandas as pd
+import pytest
+
+from windpowerlib.data import (
+    check_data_integrity,
+    check_imported_data,
+    get_turbine_types,
+    store_turbine_data_from_oedb,
+)
+
+
+class TestDataCheck:
+    @classmethod
+    def setup_class(cls):
+        cls.path = os.path.join(os.path.dirname(__file__), "oedb")
+        cls.filename = os.path.join(cls.path, "{0}.csv")
+        cls.df = pd.read_csv(cls.filename.format("turbine_data"), index_col=0)
+        cls.time_stamp = datetime.now().strftime("%Y%m%d%H%M%S")
+        cls.broken_fn = os.path.join(cls.path, "power_curves_broken.csv")
+        cls.backup_fn = os.path.join(cls.path, "power_curves_backup.csv")
+        cls.orig_path = os.path.join(
+            os.path.dirname(__file__), os.pardir, "windpowerlib", "oedb"
+        )
+        cls.orig_fn = os.path.join(cls.orig_path, "power_curves.csv")
+
+    @classmethod
+    def teardown_class(cls):
+        cls.path = os.path.join(os.path.dirname(__file__), "oedb")
+        cls.backup_fn = os.path.join(cls.path, "power_curves_backup.csv")
+        orig_path = os.path.join(
+            os.path.dirname(__file__), os.pardir, "windpowerlib", "oedb"
+        )
+        cls.orig_fn = os.path.join(orig_path, "power_curves.csv")
+        copyfile(cls.backup_fn, cls.orig_fn)
+        for f in os.listdir(cls.path):
+            if "error" in f:
+                os.remove(os.path.join(cls.path, f))
+
+    def test_normal_data_check(self):
+        copyfile(
+            self.filename.format("turbine_data"),
+            self.filename.format("turbine_data_{0}".format(self.time_stamp)),
+        )
+        for curve_type in ["power_curve", "power_coefficient_curve"]:
+            copyfile(
+                self.filename.format("{}s".format(curve_type)),
+                self.filename.format(
+                    "{0}s_{1}".format(curve_type, self.time_stamp)
+                ),
+            )
+        check_imported_data(self.df, self.filename, self.time_stamp)
+
+    def test_data_check_logging_warnings(self, caplog):
+        self.df.loc["GE158/4800", "has_power_curve"] = True
+        self.df.loc["GE100/2750", "has_cp_curve"] = True
+        check_data_integrity(self.df, min_pc_length=26)
+        assert "E48/800: power_curve is too short (25 values)" in caplog.text
+        assert "GE158/4800: No power curve" in caplog.text
+        assert "GE100/2750: No cp-curve but has_cp_curve" in caplog.text
+
+    def test_global_error(self):
+        msg = "Must have equal len keys"
+        with pytest.raises(ValueError, match=msg):
+            self.df.loc["GE158/4800", "has_cp_curve"] = [5, 3]
```

Comment:
The ValueError is already risen in this line:
`self.df.loc["GE158/4800", "has_cp_curve"] = [5, 3]`
I couldn't fix the test as I don't fully understand what should be tested.